outport/firehose/firehoseIndexerTxPool.go

```diff
@@ -0,0 +1,169 @@
+package firehose
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/firehose"
+	outportcore "github.com/ElrondNetwork/elrond-go-core/data/outport"
+	"github.com/ElrondNetwork/elrond-go-core/data/receipt"
+	"github.com/ElrondNetwork/elrond-go-core/data/rewardTx"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+)
+
+type txPool struct {
+	transactions        map[string]*firehose.TxWithFee
+	smartContractResult map[string]*firehose.SCRWithFee
+	rewards             map[string]*rewardTx.RewardTx
+	receipts            map[string]*receipt.Receipt
+	invalidTxs          map[string]*firehose.TxWithFee
+	logs                map[string]*transaction.Log
+}
+
+func getTxPool(transactionsPool *outportcore.Pool) (*txPool, error) {
+	if transactionsPool == nil {
+		return &txPool{}, nil
```

Comment:
So if the given transactionPool is nil we will return an empty txPool DTO?

Reply:
Yes, that's desired. Wanted a logic to either fill the fields or data, or, if none available -> nil

Reply:
the indexing could be inconsistent: the header should have transactions but because the transactionsPool is nil, the txs, scrs etc, will not be indexed. Is there no way to verify the consistency?

Reply:
I don't think there could be inconsistencies here, since txs are saved in outport driver from current saved block:

```
return &outportcore.Pool{
   Txs:      WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock)),
   Scrs:     WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock)),
   Rewards:  WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.RewardsBlock)),
   Invalid:  WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.InvalidBlock)),
   Receipts: WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.ReceiptBlock)),
   Logs:     odp.txCoordinator.GetAllCurrentLogs(),
}
```

Reply:
However, as discussed, changed logic to return error in case of nil tx pool